content/2018-10-31-xsimd.rst

```diff
@@ -0,0 +1,282 @@
+Bye bye boost.simd, welcome xsimd
+#################################
+
+:date: 2018-10-31
+:category: engineering
+:lang: en
+:authors: serge-sans-paille
+:summary: After five years of friendship, Pythran replaces ``boost.simd`` by
+          ``xsimd`` as its vectorization engine. This post goes through the
+          motivation of this change, and the advantage of this move.
+
+`boost.simd <https://github.com/NumScale/boost.simd>`_ provides a C++
+abstraction of vector type, allowing for efficient vectorization of array
+computations. It has been (optionally) used as part of the expression template
+engine of Pythran for a long time, a great collaboration that led to several
+patches in boost.simd, and great performance for Pythran.
+
+Unfortunately, the project has been silent over the last months (see for
+instance `this issue <https://github.com/NumScale/boost.simd/issues/546>`_) and
+I had to maintain a few custom patches. Turns out the project has been
+re-branded as **bSIMD** with a more restrictive license, as detailed in `another
+issue <https://github.com/NumScale/boost.simd/issues/545>`_. From the Pythran
+perspective, this is no good news.
+
+Fortunately, the people from `QuantStack <http://quantstack.net/>`_ have put
+tremendous effort into providing an equivalent to boost.simd, `xsimd <http://quantstack.net/xsimd.html>`_. And
+their library actually provides some improvements in the context of Pythran, so
+when they proposed to fund the move to *xsimd*, it was just perfect.
+
+So here is the deal: I do the port, report any API and/or performance issue,
+and eventually provide patches when relevant. That's what I did over the last
+three months, let's have a look at the results.
+
+User-level Changes
+==================
+
+In order to activate explicit vectorisation, one must pass ``-DUSE_XSIMD -march=native`` to the Pythran compiler, in place of ``-DUSE_BOOST_SIMD -march=native``. Fair enough.
```

Comment:
This might be a bit off-topic, but why not change the SIMD flag to something generic, like `-DUSE_EXPLICIT_SIMD`